examples/chol.dx

```diff
@@ -0,0 +1,104 @@
+' # Cholesky Factorization
+https://en.wikipedia.org/wiki/Cholesky_decomposition
+
+' ## Helpers
+
+ixkey2:: A n::Ix m::Ix. Key -> n -> m -> Key
+ixkey2 x i = ixkey $ ixkey x i
+
+' ### Index checking
+
+inUpper:: A m::Ix. m -> m -> Bool
+inUpper i j = (asint i) < (asint j)
+
+inLower:: A m::Ix. m -> m -> Bool
+inLower i j = (asint i) > (asint j)
+
+inDiag:: A m::Ix. m -> m -> Bool
+inDiag i j = (asint i) == (asint j)
+
+' ### Matrix Math
+
+vdot :: n=>Real -> n=>Real -> Real
+vdot x y = sum for i. x.i * y.i
+
+
+mmp :: l=>m=>Real -> m=>n=>Real -> l=>n=>Real
+mmp m1 m2 = for i k. sum for j. m1.i.j * m2.j.k
+
+
+eyeLike:: n=>n=>Real -> n=>n=>Real
+eyeLike x = for i. for j.
+  select (inDiag i j) 1.0 0.0
+
+transpose::n=>m=>Real->m=>n=>Real
+transpose x = for i. for j. x.j.i
+
+mmadd:: (n=>m=>Real)->(n=>m=>Real)->(n=>m=>Real)
+mmadd x y = for i. for j. x.i.j + y.i.j
+
+
+
+' ## Cholesky Algorithm
+
+chol :: (n=>n=>Real) -> (n=>n=>Real)
+chol x = ..
+  diagOp r i = 
+    innersum = sum for k::n. ..
+      select (inLower k i) (sq r.i.k) 0.0
```

Comment:
This should be:
`select (inLower i k) (sq r.i.k) 0.0`

Reply:
Did that fix it?

Reply:
Not tested yet.

Reply:
that swap makes everything NaN, except first row.
But it is still correct by the formula AFAICT